exec/internal/compile/backend_amd64.go

```diff
@@ -0,0 +1,311 @@
+// Copyright 2019 The go-interpreter Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package compile
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	ops "github.com/go-interpreter/wagon/wasm/operators"
+	asm "github.com/twitchyliquid64/golang-asm"
+	"github.com/twitchyliquid64/golang-asm/obj"
+	"github.com/twitchyliquid64/golang-asm/obj/x86"
+)
+
+// dirtyRegs hold booleans that are true when the register stores
+// a reserved value that needs to be flushed to memory.
+type dirtyRegs struct {
+	R12 bool
+	R13 bool
+}
+
+// Details of the AMD64 backend:
```

Comment:
are those details inherited from the Go AMD64 backend or just how we decided to wield it?

Reply:
Just how I decided to wield it right now. I plan on changing this up to be more dynamic / efficient in the future - this is just a naive first implementation.

(In particular, most of the instructions emitted around handling go slices could be omitted, and intermediate values between operations could be stored in registers rather than always pushed to the wasm stack).

Reply:
ok. thanks.